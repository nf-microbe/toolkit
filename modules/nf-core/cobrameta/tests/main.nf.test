nextflow_process {

    name "Test Process COBRAMETA"
    script "../main.nf"
    process "COBRAMETA"

    tag "modules"
    tag "modules_nfcore"
    tag "cobrameta"


    test("b_fragilis - spades") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_coverage.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_names.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1_contigs.sorted.bam', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/illumina/log/test1.spades.log", checkIfExists: true)
                ]
                input[5] = 'spades'
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out.fasta,
                    file(process.out.log[0][1]).name,
                    process.out.joining_summary,
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("sarscov2 - megahit") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_coverage.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_names.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1_contigs.sorted.bam', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/sarscov2/illumina/log/test.megahit.log", checkIfExists: true)
                ]
                input[5] = 'megahit'
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out.fasta,
                    file(process.out.log[0][1]).name,
                    process.out.joining_summary,
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_coverage.tsv", checkIfExists: true)
                ]
                input[2] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/text/test1_contig_names.tsv", checkIfExists: true)
                ]
                input[3] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1_contigs.sorted.bam', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'test'],
                    file("${projectDir}/assets/test-datasets/genomics/sarscov2/illumina/log/test.megahit.log", checkIfExists: true)
                ]
                input[5] = 'megahit'
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
