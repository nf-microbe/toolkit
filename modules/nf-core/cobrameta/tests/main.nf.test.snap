{
    "sarscov2 - megahit": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_COBRA_extended.fasta.gz:md5,3ee264adad01d595157f023506a68248"
                ]
            ],
            "test.cobra.log",
            [
                [
                    {
                        "id": "test"
                    },
                    "test_COBRA_joining_summary.txt:md5,fc51a413c8765bc59954f9a359e7a0f2"
                ]
            ],
            [
                "versions.yml:md5,6830d6d3da149b4eebd6e66f2346ec92"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T12:45:23.54931085"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_COBRA_extended.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_COBRA_joining_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cobra.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,6830d6d3da149b4eebd6e66f2346ec92"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_COBRA_extended.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "joining_summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test_COBRA_joining_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cobra.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6830d6d3da149b4eebd6e66f2346ec92"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-04T08:12:21.739882401"
    },
    "b_fragilis - spades": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_COBRA_extended.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                ]
            ],
            "test.cobra.log",
            [
                [
                    {
                        "id": "test"
                    },
                    "test_COBRA_joining_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,6830d6d3da149b4eebd6e66f2346ec92"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-23T11:12:12.708997415"
    }
}